### PR TITLE
[MRG] add citation output

### PR DESCRIPTION
**Pull request by ctb**
Per #259. #478 is probably a better solution, but this is complementary?

- [x] Is it mergeable?
- [x] `make test` Did it pass the tests?
- [x] `make coverage` Is the new code covered?
- [x] Did it change the command-line interface? Only additions are allowed
  without a major version increment. Changing file formats also requires a
  major version number increment.
- [x] Was a spellchecker run on the source code and documentation after
  changes were made?

### CODE REVIEW

**Comment by ctb** [comment]
@luizirber your thoughts welcome on this approach. All that is left to do for 2.0.0, I think, is to replace all of the end-point ArgumentParser objects with the new class.

**Comment by luizirber** [comment]
I like it. #478 is more general, but for now it is more complicated to implement and I don't want to see it blocking 2.0

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,6 @@
+cff-version: 1.0.3
+message: If you use this software, please cite it as below.
+title: sourmash: a library for MinHash sketching of DNA
+version: 2.0.0
+doi: 10.21105/joss.00027
+date-released: 2019-01-10
```
Comment (luizirber) [inline CITATION.cff]: Do you want to add authors here?
```suggestion
authors:
  - family-names: Brown
    given-names: C. Titus
    orcid: https://orcid.org/0000-0001-6001-2677
  - family-names: Irber Junior
    given-names: Luiz Carlos
    orcid: https://orcid.org/0000-0003-4371-9659
```

Comment (ctb) [reply CITATION.cff]: Sure, but I expect us to swiftly expand the author list in favor of a JOSS paper for sourmash 2.0 :)

**Comment by ctb** [comment]
Ready for review @luizirber

**Pull request merged**